pkg/controller/goldmane/controller.go

```diff
@@ -236,3 +244,45 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 
 	return reconcile.Result{}, nil
 }
+
+// maintainInstallationFinalizer manages this controller's finalizer on the Installation resource.
+// We add a finalizer to the Installation when Goldmane has been installed, and only remove that finalizer when
+// the Goldmane has been deleted and its pods have stopped running. This allows for a graceful cleanup of Goldmane resources
+// prior to the CNI plugin being removed.
+func maintainInstallationFinalizer(ctx context.Context, c client.Client, goldmaneCr *operatorv1.Goldmane) error {
+	// Get the Installation.
+	installation := &operatorv1.Installation{}
+	if err := c.Get(ctx, utils.DefaultInstanceKey, installation); err != nil {
+		if errors.IsNotFound(err) {
+			log.V(1).Info("Installation config not found")
+			return nil
+		}
+		log.Error(err, "An error occurred when querying the Installation resource")
+		return err
+	}
+	patchFrom := client.MergeFrom(installation.DeepCopy())
+
+	// Determine the correct finalizers to apply to the Installation. If the Whisker exists, we should apply
+	// a finalizer. Otherwise, if the Whisker namespace doesn't exist we should remove it. This ensures the finalizer
+	// is always present so long as the resources managed by this controller exist in the cluster.
+	if goldmaneCr != nil {
+		// Add a finalizer indicating that the Goldmane is still running.
+		utils.SetInstallationFinalizer(installation, render.GoldmaneFinalizer)
+	} else {
+		// Check if the Goldmane namespace exists, and remove the finalizer if not. Gating this on Namespace removal
+		// in the best way to approximate that all Goldmane related resources have been removed.
+		l := &corev1.Namespace{}
+		err := c.Get(ctx, types.NamespacedName{Name: goldmane.GoldmaneNamespace}, l)
```

Comment:
Same here - there is no Goldmane namespace, it runs within `calico-system` so we can't use the same heuristic here.